### DISCUSSION TITLE
Assignment of high dimensional sdp array

### DISCUSSION THREAD

**Discussion by BYS543** [yalmip/YALMIP#1101] [Problems using YALMIP] [answered]
Hi Johan,
```
A = sdpvar(3,4,1);
B = sdpvar(3,4,1);
A(2,:,1)=B(2,:,1);
```
why it tells me:`N-dimensional indexing allowed for Full matrices only`.
How can i deal with it?

**Comment by johanlofberg** [comment, marked as answer]
When the size of the highest dimension is 1, the variable is automatically collapsed to a lower dimension

**Reply by BYS543** [reply]
Thank you for your reply, Johan. I see.
